internal/executor/service/job_lease.go

```diff
@@ -136,16 +145,49 @@ func (jobLeaseService *JobLeaseService) renewJobLeases(pods []*v1.Pod) {
 	}
 }
 
-func getFinishedPods(pods []*v1.Pod) []*v1.Pod {
-	finishedPods := make([]*v1.Pod, 0)
-
+func (jobLeaseService *JobLeaseService) markAsDone(pods []*v1.Pod) {
 	for _, pod := range pods {
-		if isPodReadyForCleanup(pod) {
-			finishedPods = append(finishedPods, pod)
+		err := jobLeaseService.clusterContext.AddAnnotation(pod, map[string]string{
+			jobDoneAnnotation: time.Now().String(),
+		})
+		if err != nil {
+			log.Warnf("Failed to annotate pod as done: %v", err)
 		}
 	}
+}
 
-	return finishedPods
+func isToRenew(pod *v1.Pod) bool {
+	return !isReportedDone(pod)
+}
+
+func isToReportDone(pod *v1.Pod) bool {
```

Comment:
Maybe isReadyToBeReportedDone? Or something. Not sure, isToReportDone sounds odd to me